zcm/transport/transport_file.cpp

```diff
@@ -115,15 +115,15 @@ struct ZCM_TRANS_CLASSNAME : public zcm_trans_t
         return ZCM_EOK;
     }
 
-    int recvmsg(zcm_msg_t *msg, int timeout)
+    int recvmsg(zcm_msg_t *msg, unsigned timeout)
     {
         assert(mode == "r");
         if (!good()) {
             // TODO Build in a way for a transport to tell zcm that an "error"
             //      has occurred. Not sure what to do here since this function
             //      has no way of communicating to the caller that this function
             //      shouldn't be called anymore
-            usleep(timeout);
+            if (timeout > 0) usleep(timeout);
```

Comment:
Found another bug:

timeout is in millis 
usleep() is in micros

Reply:
yeah good catch. will fix